homeassistant/components/imap_email_content/repairs.py

```diff
@@ -0,0 +1,170 @@
+                    {
+                        "id": "custom_event",
+                        "platform": "event",
+                        "event_type": "imap_content",
+                        "event_data": {"sender": config[CONF_SENDERS][0]},
+                    }
+                ],
+                "sensor": [
+                    {
+                        "state": template,
+                        "name": name,
+                    }
+                ],
+            }
+        ]
+    }
+
+    data = {
+        CONF_SERVER: config[CONF_SERVER],
+        CONF_PORT: config[CONF_PORT],
+        CONF_USERNAME: config[CONF_USERNAME],
+        CONF_PASSWORD: config[CONF_PASSWORD],
+        CONF_FOLDER: config[CONF_FOLDER],
+    }
+    data[CONF_VALUE_TEMPLATE] = template
+    data[CONF_NAME] = name
+    placeholders = {"yaml_example": yaml.dump(template_sensor_config)}
+    placeholders.update(data)
+
+    ir.async_create_issue(
+        hass,
+        DOMAIN,
+        issue_id,
+        breaks_in_ha_version="2023.10.0",
+        is_fixable=True,
+        severity=ir.IssueSeverity.WARNING,
+        translation_key="migration",
+        translation_placeholders=placeholders,
+        data=data,
+    )
+
+
+class DeprecationRepairFlow(RepairsFlow):
+    """Handler for an issue fixing flow."""
+
+    def __init__(self, issue_id: str, config: ConfigType) -> None:
+        """Create flow."""
+        self._name: str = config[CONF_NAME]
+        self._config: dict[str, Any] = config
+        self._issue_id = issue_id
+        super().__init__()
+
+    async def async_step_init(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the first step of a fix flow."""
+        return await self.async_step_start()
+
+    @callback
+    def _async_get_placeholders(self) -> dict[str, str] | None:
+        issue_registry = ir.async_get(self.hass)
+        description_placeholders = None
+        if issue := issue_registry.async_get_issue(self.handler, self.issue_id):
+            description_placeholders = issue.translation_placeholders
+
+        return description_placeholders
+
+    async def async_step_start(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Wait for the user to start the config migration."""
+        placeholders = self._async_get_placeholders()
+        if user_input is None:
+            return self.async_show_form(
+                step_id="start",
+                data_schema=vol.Schema({}),
+                description_placeholders=placeholders,
+            )
+
+        return self.async_show_form(
+            step_id="confirm",
+            data_schema=vol.Schema({}),
+            description_placeholders=placeholders,
+        )
+
+    async def async_step_confirm(
+        self, user_input: dict[str, str]
+    ) -> data_entry_flow.FlowResult:
+        """Handle the confirm step of a fix flow."""
```

Comment:
`user_input` for `async_step_confirm` will never be `None` because `async_show_form` is called directly.

Reply:
We should not call the form of another step. The form is private to each step.